src/main/java/bootcamp/services/twitter4j/TwitterResourceService.java

```diff
@@ -64,6 +78,7 @@ public Tweet statusToTweet(Status status) {
         logger.info("TwitterResourceService called postTweet");
         try {
             StatusUpdate statusUpdate = new StatusUpdate(post.getMessage());
+            cache.invalidateAll(); // clear cache so that the new message is there when it's loaded again
```

Comment:
is this the best place for this line?

Reply:
I'm not sure, but I don't think it'd be good anywhere else. I wanted to make sure that the status is actually updated before clearing the cache, so that it isn't cleared "without a reason". Though I suppose, in a way, putting it before hand would allow to force a refresh of the cache without posting, by submitting an invalid post. Is that what you had in mind?

Reply:
yeah, that's exactly it the thought process on why it makes a difference

Reply:
Updated